### PR TITLE
Fix issue with date picker

### PR DESCRIPTION
**Pull request by JoyLubega**
Resolve https://github.com/opencollective/opencollective/issues/5162

# Description

# Screenshots

looam video: https://www.loom.com/share/21749b4533c4446f987f5a84c64c13e9

### CODE REVIEW

**Comment by JoyLubega** [comment]
I noticed the bug was created beacuse dayjs was formating the date before the user is done typing.

**Comment by JoyLubega** [comment]
@SudharakaP  can I get some feedback on this when you get some time. Thanks

**Review by SudharakaP — APPROVED** [review]
Tested this fix with several locales and seems to work fine. 👍🏽

**Comment by JoyLubega** [comment]
Hi @Betree  is this ready for merge, any feedback?

File: components/budget/filters/PeriodFilter.js [hunk 1]
```diff
@@ -48,10 +48,16 @@ const getIntervalFromValue = value => {
...
 
   // Reset interval in case fromDate is after toDate
-  if (newInterval.from && newInterval.to && newInterval.from > newInterval.to) {
+  if (
+    newInterval.from &&
+    !newInterval.from.startsWith('0') &&
+    newInterval.to &&
+    !newInterval.to.startsWith('0') &&
+    newInterval.from > newInterval.to
+  ) {
```
Comment (Betree) [inline components/budget/filters/PeriodFilter.js:60]: That works for me, but let's add some clues to help other developers understand what's going on here. We could add some comments, or extract that code to a clearly named helper:

```suggestion
  if (isValidDate(newInterval.from) && isValidDate(newInterval.to) && newInterval.from > newInterval.to) {
```

Comment (JoyLubega) [reply components/budget/filters/PeriodFilter.js]: Thanks @Betree  I have made the change

**Review by Betree — APPROVED** [review]
Perfect!

**Pull request merged**